pkg/operator/openshift_sdn.go

```diff
@@ -109,3 +102,95 @@ func sdnPluginName(n v1.SDNMode) string {
 	}
 	return ""
 }
+
+// controllerConfig builds the contents of controller-config.yaml
+// for the controller
+func (h *Handler) controllerConfig() (string, error) {
+	c := h.config.Spec.DefaultNetwork.OpenshiftSDNConfig
+
+	// generate master network configuration
+	ippools := []cpv1.ClusterNetworkEntry{}
+	for _, net := range h.config.Spec.ClusterNetworks {
+		ippools = append(ippools, cpv1.ClusterNetworkEntry{CIDR: net.CIDR, HostSubnetLength: net.HostSubnetLength})
+	}
+
+	var vxlanPort uint32 = 4789
+	if c.VXLANPort != nil {
+		vxlanPort = *c.VXLANPort
+	}
+
+	cfg := cpv1.OpenShiftControllerManagerConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "openshiftcontrolplane.config.openshift.io/v1",
+			Kind:       "OpenShiftControllerManagerConfig",
+		},
+		// no ObjectMeta - not an API object
+
+		Controllers: []string{"openshift.io/sdn"},
+		Network: cpv1.NetworkControllerConfig{
+			NetworkPluginName:  sdnPluginName(c.Mode),
+			ClusterNetworks:    ippools,
+			ServiceNetworkCIDR: h.config.Spec.ServiceNetwork,
+			VXLANPort:          vxlanPort,
+		},
+
+		LeaderElection: configv1.LeaderElection{
+			Namespace: "openshift-sdn",
+			Name:      "openshift-sdn-controller",
```

Comment:
Neither of these fields gets read. If we're not providing a way to default the *other* fields (timeouts or whatever they are) then you can skip the LeaderElection section)